src/mds/CDir.cc

```diff
@@ -2191,70 +2342,56 @@ void CDir::_omap_commit(int op_prio)
     // fnode.snap_purged_thru = realm->get_last_destroyed();
   }
 
-  set<string> to_remove;
-  map<string, bufferlist> to_set;
+  vector<string> _stale_items;
+  _stale_items.reserve(stale_items.size());
 
-  C_GatherBuilder gather(g_ceph_context,
-			 new C_OnFinisher(new C_IO_Dir_Committed(this,
-								 get_version()),
-					  mdcache->mds->finisher));
+  vector<dentry_key_t> to_remove;
+  to_remove.reserve(10);
 
-  SnapContext snapc;
-  object_t oid = get_ondisk_object();
-  object_locator_t oloc(mdcache->mds->mdsmap->get_metadata_pool());
+  vector<dentry_commit_items> to_set;
+  to_set.reserve(10);
```

Comment:
count items and reserve enough at fist

Reply:
Sure, will fix it.

Reply:
@ukernel 

BTW, for the `to_set` we need to count the `dirty_dentries`, which is the `elist<>` type, I am thinking whether will it make sense to count this in for loop ? Or should we support the `elist::size()` helper first ?

Reply:
do it in separate PR if you want

Reply:
Sure, will try to add the `elist::size()` member function in a separate PR.

Reply:
Please see https://github.com/ceph/ceph/pull/37151